lib/modules/manager/pep621/schema.ts

```diff
@@ -45,7 +46,13 @@ const UvSource = z.object({
 
 const UvSchema = z.object({
   'dev-dependencies': DependencyListSchema,
-  sources: z.record(z.string(), UvSource).optional(),
+  sources: z
+    .record(
+      // uv applies the same normalization as for Python dependencies on sources
+      z.string().transform((source) => normalizePythonDepName(source)),
```

Comment:
Note on this: if we have 2 sources with different casing that would actually be normalised to the same name, Renovate will only keep the 2nd one. This is in line with what uv does today, which will likely be fixed to avoid this case anyway (https://github.com/astral-sh/uv/issues/7229).